R/predict_setting_contacts.R

```diff
@@ -2,26 +2,53 @@
 #' @param population population
 #' @param contact_model contact_model
 #' @param age_breaks age_breaks
-#' @return list of matrices
+#' @param per_capita_household_size Optional (defaults to NULL). When set, it 
+#'   adjusts the household contact matrix by some per capital household size. 
+#'   To set it, provide a single number, which is the per capita household size. 
+#'   See `get_per_capita_household_size()` function for a helper for Australian
+#'    data with a workflow on how to ge this number.
+#' @param ... extra arguments to pass to the `per_capita_household_size` 
+#'     adjustment - see `adjust_household_contact_matrix` for details. 
+#'     Default values for this are `household_contact_rate = 1` and 
+#'     `model_mean_other_householders = 2.248971`.
+#' @return List of setting matrices
 #' @author Nicholas Tierney
 #' @export
 #' @examples 
 #' # don't run as it takes too long to fit
 #' \dontrun{
-#' contact_model <- fit_setting_contacts(
-#'   contact_data_list = get_polymod_setting_data(), 
-#'   survey_population = get_polymod_population()
+#' fairfield_age_pop <- abs_age_lga("Fairfield (C)")
+#' fairfield_age_pop
+#' 
+#' polymod_contact_data <- get_polymod_setting_data()
+#' polymod_survey_data <- get_polymod_population()
+#' 
+#' setting_models <- fit_setting_contacts(
+#'   contact_data_list = polymod_contact_data,
+#'   population = polymod_survey_data
+#' )
+#' 
+#' synthetic_settings_5y_fairfield <- predict_setting_contacts(
+#'   population = fairfield_age_pop,
+#'   contact_model = setting_models,
+#'   age_breaks = c(seq(0, 85, by = 5), Inf)
 #' )
 #' 
-#' contact_model
+#' fairfield_hh_size <- get_per_capita_household_size(lga = "Fairfield (C)")
+#' fairfield_hh_size
 #' 
-#' contact_model_pred <- predict_setting_contacts(
-#'   population = get_polymod_population(), 
-#'   contact_model = contact_model, 
-#'   age_breaks = c(seq(0, 75, by = 5), Inf)
+#' synthetic_settings_5y_fairfield_hh <- predict_setting_contacts(
+#'   population = fairfield_age_pop,
+#'   contact_model = setting_models,
+#'   age_breaks = c(seq(0, 85, by = 5), Inf),
+#'   per_capita_household_size = fairfield_hh_size
 #' )
 #' }
-predict_setting_contacts <- function(population, contact_model, age_breaks) {
+predict_setting_contacts <- function(population, 
+                                     contact_model, 
+                                     age_breaks,
+                                     per_capita_household_size = NULL,
+                                     ...) {
```

Comment:
suggest we replace the dots with an explicit user-facing argument:
```
model_percapita_household_size
```
with default:
```
model_percapita_household_size = get_polymod_percapita_household_size()
```
and this can then get passed through internally. The other argument `household_contact_rate` is no longer needed